fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -0,0 +1,338 @@
+            } else {
+                throw new IllegalArgumentException("Unsupported join criteria");
+            }
+        }
+
+        JoinRelation joinRelation = new JoinRelation(joinType, left, right, predicate, context.LATERAL() != null);
+        joinRelation.setUsingColNames(usingColNames);
+        if (context.hint() != null) {
+            joinRelation.setJoinHint(context.hint().IDENTIFIER(0).getText());
+        }
+
+        return joinRelation;
+    }
+
+    @Override
+    public ParseNode visitSetOperation(StarRocksParser.SetOperationContext context) {
+        QueryRelation left = (QueryRelation) visit(context.left);
+        QueryRelation right = (QueryRelation) visit(context.right);
+
+        boolean distinct = true;
+        if (context.setQuantifier() != null) {
+            if (context.setQuantifier().DISTINCT() != null) {
+                distinct = true;
+            } else if (context.setQuantifier().ALL() != null) {
+                distinct = false;
+            }
+        }
+
+        SetQualifier setQualifier = distinct ? SetQualifier.DISTINCT : SetQualifier.ALL;
+        switch (context.operator.getType()) {
+            case StarRocksLexer.UNION:
+                if (left instanceof UnionRelation && ((UnionRelation) left).getQualifier().equals(setQualifier)) {
+                    ((UnionRelation) left).addRelation(right);
+                    return left;
+                } else {
+                    return new UnionRelation(Lists.newArrayList(left, right), setQualifier);
+                }
+            case StarRocksLexer.INTERSECT:
+                if (left instanceof IntersectRelation &&
+                        ((IntersectRelation) left).getQualifier().equals(setQualifier)) {
+                    ((IntersectRelation) left).addRelation(right);
+                    return left;
+                } else {
+                    return new IntersectRelation(Lists.newArrayList(left, right), setQualifier);
+                }
+            case StarRocksLexer.EXCEPT:
+                if (left instanceof ExceptRelation && ((ExceptRelation) left).getQualifier().equals(setQualifier)) {
+                    ((ExceptRelation) left).addRelation(right);
+                    return left;
+                } else {
+                    return new ExceptRelation(Lists.newArrayList(left, right), setQualifier);
+                }
+        }
+        throw new IllegalArgumentException("Unsupported set operation: " + context.operator.getText());
+    }
+
+    @Override
+    public ParseNode visitInlineTable(StarRocksParser.InlineTableContext context) {
+        List<ValueList> rowValues = visit(context.rowConstructor(), ValueList.class);
+        List<ArrayList<Expr>> rows = rowValues.stream().map(ValueList::getFirstRow).collect(toList());
+
+        List<String> colNames = new ArrayList<>();
+        for (int i = 0; i < rows.get(0).size(); ++i) {
+            colNames.add("column_" + i);
```

Comment:
Is it possible that two rows consists of different columns ? In which the first row could not be used as label ?

Reply:
two rows consists of different columns will throw Semantic Exception in Analyzer

Reply:
What if the column has default value? The first row miss some columns, and the second row contains all columns?

Reply:
There is no such situation, here it means select xx from values()